File: app/src/main/java/com/example/gitapp/MainActivity.java
package com.example.gitapp;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {
    EditText editText;
    Button button;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        editText=findViewById(R.id.username);
        button=findViewById(R.id.button);

    }

    public void onclick(View view) {
        Intent intent = new Intent(MainActivity.this,UserActivity.class);
        intent.putExtra("username",editText.getText().toString());
        startActivity(intent);

    }
}

File: app/src/main/java/com/sid/github/activity/MainActivity.kt
package com.sid.github.activity

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import com.google.android.material.textfield.TextInputEditText
import com.google.android.material.textfield.TextInputLayout
import com.sid.github.R

class MainActivity : AppCompatActivity() {

    lateinit var username : TextInputEditText
    lateinit var search : Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        username = findViewById(R.id.textInput)
        search = findViewById(R.id.btnSearch)

        search.setOnClickListener {
            if(username.text?.isEmpty() == true){
                username.error = "Mandatory"
                username.requestFocus()
                return@setOnClickListener
            }
            val intent = Intent(this, ProfileActivity::class.java)
            intent.putExtra("USERNAME",username.text.toString())
            username.text?.clear()
            startActivity(intent)
        }

    }
}